src/sentry/static/sentry/app/components/events/interfaces/utils.jsx

```diff
@@ -30,10 +29,15 @@ export function getCurlCommand(data) {
     result += ' \\\n -H "' + header[0] + ': ' + escapeQuotes(header[1] + '') + '"';
   }
 
-  if (_.isString(data.data)) {
-    result += ' \\\n --data "' + escapeQuotes(data.data) + '"';
-  } else if (defined(data.data)) {
-    result += ' \\\n --data "' + escapeQuotes(jQuery.param(data.data)) + '"';
+  switch (data.inferredContentType) {
+    case 'application/json':
```

Comment:
nit, but would probably be nice to only switch on setting the value of a var called dataString, and do
` result += ' \\\n --data "' + escapeQuotes(dataString)'"'`